maintenance-poller/azure_metadata_client.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+)
+
+type MetadataClient interface {
+	GetScheduledEvents() (ScheduledEvent, error)
+	ConfirmScheduledEvent(eventId string) (statusCode int, err error)
+}
+
+type AzureMetadataClient struct {
+	client http.Client
+}
+
+func NewAzureMetadataClient() *AzureMetadataClient {
+	c := AzureMetadataClient{}
+	c.client = http.Client{}
+	return &c
+}
+
+func (c AzureMetadataClient) GetScheduledEvents() (ScheduledEvent, error) {
+	scheduledEvent := ScheduledEvent{}
+
+	req, err := http.NewRequest("GET", "http://169.254.169.254/metadata/scheduledevents", nil)
+	if err != nil {
+		return scheduledEvent, err
+	}
+
+	q := req.URL.Query()
+	q.Add("api-version", "2020-07-01")
+	req.URL.RawQuery = q.Encode()
+
+	req.Header.Add("Metadata", "true")
+
+	res, err := c.client.Do(req)
+	if err != nil {
+		return scheduledEvent, err
+	}
+
+	defer res.Body.Close()
+
+	err = json.NewDecoder(res.Body).Decode(&scheduledEvent)
+
+	return scheduledEvent, err
+}
+
+func (c AzureMetadataClient) ConfirmScheduledEvent(eventId string) (statusCode int, err error) {

```

Comment:
it would be nice to have function level comments in all functions: https://tip.golang.org/doc/comment